### PR TITLE
added algorithm for grayscale filter

### PR DESCRIPTION
**Pull request by sabyasachi07**

### CODE REVIEW

**Comment by chandrakant100** [comment]
please resolve merge conflicts

File: helpers.cpp [hunk 1]
```diff
@@ -8,7 +8,22 @@
...
+    
+    // create temporary value to store the calculations
+    int avg {0};
+    
+    // Iterating over each pixel
+    for (int i {0}; i < height; ++i) {
+        for (int j {0}; j < width; ++j) {
+            
+            // averages the color intensity and then applies the same value to all the colors to get gray
+            avg = round((image[i][j].rgbtBlue + image[i][j].rgbtGreen + image[i][j].rgbtRed) / 3.0);
```
Comment (Himanshu40) [inline helpers.cpp:20]: use static_cast here to convert double 3.0 to int.

File: helpers.cpp [hunk 2]
```diff
@@ -8,7 +8,22 @@
...
+
+            image[i][j].rgbtBlue = avg;
+            image[i][j].rgbtGreen = avg;
+            image[i][j].rgbtRed = avg;
```
Comment (Himanshu40) [inline helpers.cpp:24]:
```suggestion
            image[i][j].rgbtBlue = avg;
            image[i][j].rgbtGreen = avg;
            image[i][j].rgbtRed = avg;
```
Add proper spacings.

**Review by Himanshu40 — CHANGES_REQUESTED** [review]

File: helpers.cpp (hunk 2 above)
Comment (sabyasachi07) [reply helpers.cpp:24]: those are properly spaced , I formatted the code with clang before I pushed so you check again with your script

File: helpers.cpp (hunk 1 above)
Comment (chandrakant100) [reply helpers.cpp:20]: do static cast is needed here because here avg is declared and initialize above.

**Comment by chandrakant100** [comment]
I have Closed this Pr, Kindly make a new fork, Create branch and push your changes.

**Pull request closed without merging**